lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -82,15 +82,15 @@ void _hideAutofillElements(html.HtmlElement domElement) {
 class EngineAutofillForm {
   EngineAutofillForm({this.formElement, this.elements, this.items});
 
-  final html.FormElement formElement;
+  final html.FormElement? formElement;
 
-  final Map<String, html.HtmlElement> elements;
+  final Map<String, html.HtmlElement>? elements;
 
-  final Map<String, AutofillInfo> items;
+  final Map<String, AutofillInfo>? items;
 
-  factory EngineAutofillForm.fromFrameworkMessage(
-    Map<String, dynamic> focusedElementAutofill,
-    List<dynamic> fields,
+  static EngineAutofillForm? fromFrameworkMessage(
```

Comment:
question: why do we need to change the factory? (I am asking to learn cause I didn't tried NNB changes myself)

Reply:
I'm curious too.

Reply:
This is a new language restriction. Factories are no longer allowed to return `null` (otherwise you can't do `Foo foo = Foo();` if you are calling a factory). I was able to de-null some of our factories, but not all, so I just made it a static method. Since there's no `new` keyword in Dart any more the difference is minor.